### PR TITLE
Patchqueue property tests

### PR DESCRIPTION
**Pull request by euanh**
This pull request adds some Hypothesis property-based tests for patchqueue guard handling.  Guard handling was already tested by point test in test_patchqueues.py so coverage reported by nosetests does not change, but the property-based tests explore some corner cases which the point test does not.

### CODE REVIEW

File: tests/test_patchqueue.py [hunk 1]
```diff
@@ -27,11 +27,11 @@ def tearDown(self):
         shutil.rmtree(self.test_dir)
 
     def test_patch_guards(self):
+        """Patch guards are applied correctly"""
         series = ("patch_no_guard_or_comment\n",
                   "patch_with_a_comment # this is a comment\n",
                   "patch_with_a_positive_guard #+aguard\n",
-                  "patch_with_a_negative_guard #-aguard\n",
-                  )
```
Comment (mseri) [inline tests/test_patchqueue.py]: If you leave the parenthesis on a new line, adding a new patch will not require two line changes.

File: tests/test_patchqueue.py [hunk 2]
```diff
@@ -50,13 +51,103 @@ def test_rewrite_spec(self):
         self.assertIn("Patch2: third.patch\n", rewritten)
 
     def test_autosetup_present(self):
+        """Patchqueue application succeeds if %autosetup is present"""
         spec = Spec("tests/data/manifest/branding-xenserver.spec",
                     check_package_name=False)
         patches = ["first.patch"]
-        planex.patchqueue.expand_patchqueue(spec, patches)
```
Comment (mseri) [inline tests/test_patchqueue.py]: :O Was anything tested here before this PR?

**Comment by mseri** [comment]
The strategies are impressively readable. We should start using hypothesis more.

**Comment by mseri** [comment]
Feel free to merge or to add a newline before the parenthesis (see comment above)

File: tests/test_patchqueue.py (hunk 1 above)
Comment (euanh) [inline tests/test_patchqueue.py]: Yes, but this seems to be the style we use elsewhere.   (I noticed this because pylint complained that the closing parenthesis was wrongly indented.)

File: tests/test_patchqueue.py (hunk 2 above)
Comment (euanh) [inline tests/test_patchqueue.py]: Yes, this test was verifying that expand_patchqueue() did not raise an exception.   However pylint complained that the test did not use self and therefore didn't need to be a member.   This change seemed better than just saying 'self.assertTrue(True)' to shut pylint up...
